canopy.py

```diff
@@ -361,7 +411,66 @@ def correct_inverted_canopy_tiff(inverted_phyreg_ids):
                 # copy raster is used as arcpy.save does not give bit options.
                 arcpy.CopyRaster_management(corrected, corrected_path,
-                        pixel_type='2_BIT')
+                                            nodata_value = '3',
+                                            pixel_type='2_BIT')
+
+    # clear selection
+    arcpy.SelectLayerByAttribute_management(phyregs_layer, 'CLEAR_SELECTION')
+
+    print('Completed')
+
+def convert_canopy_tif_to_shp(phyreg_id):
+    '''
+    This function converts the canopy TIFF files to shapefile. If a region
+    has been corrected for inverted values the function will convert the
+    corrected TIFF to shapefile instead of the original canopy TIFF. If no
+    corrected TIFF exists for a region then the original canopy TIFF will be
+    converted.
+
+    phyreg_ids: list of physiographic region IDs to process
+    '''
+    phyregs_layer = canopy_config.phyregs_layer
+    analysis_year = canopy_config.analysis_year
+    snaprast_path = canopy_config.snaprast_path
+    results_path = canopy_config.results_path
+
+    arcpy.env.addOutputsToMap = False
+    arcpy.env.snapRaster = snaprast_path
+
+    arcpy.SelectLayerByAttribute_management(phyregs_layer,
+            where_clause='PHYSIO_ID in (%s)' % ','.join(
+                map(str, phyreg_id)))
+    with arcpy.da.SearchCursor(phyregs_layer, ['NAME', 'PHYSIO_ID']) as cur:
+        for row in cur:
+            name = row[0]
+            print(name)
+            name = name.replace(' ', '_').replace('-', '_')
+            phyreg_id = row[1]
+            outdir_path = '%s/%s/Outputs' % (results_path, name)
+            if not os.path.exists(outdir_path):
+                continue
+            canopytif_path = '%s/canopy_%d_%s.tif' % (outdir_path,
+                                                      analysis_year, name)
+            corrected_path = '%s/corrected_canopy_%d_%s.tif' % (
+                outdir_path, analysis_year, name)
+            # Add shp_ as prefix to output shapefile
+            canopyshp_path = '%s/shp_canopy_%d_%s.shp' % (
+                outdir_path, analysis_year, name)
+            if os.path.exists(canopyshp_path):
+                continue
+            if not os.path.exists(canopyshp_path):
+                # Check for corrected inverted TIFF first
+                if os.path.exists(corrected_path):
+                    # Do not simplify polygons, keep cell extents
+                    arcpy.RasterToPolygon_conversion(corrected_path,
+                            canopyshp_path, 'NO_SIMPLIFY', 'Value')
+                    continue
+                # If no corrected inverted TIFF use orginial canopy TIFF
+                elif os.path.exists(canopytif_path):
+                    # Do not simplify polygons, keep cell extents
+                    arcpy.RasterToPolygon_conversion(canopytif_path,
+                            canopyshp_path, 'NO_SIMPLIFY', 'Value')
+                    continue
```

Comment:
Please rename the `grid_code` field to `Canopy` and delete the default `Id` field.